### ISSUE TITLE
The vm should be connectable after az vm start process has finished

### ISSUE THREAD

**Issue by Lelelo1** [Azure/azure-cli#10017]
**Is your feature request related to a problem? Please describe.**

**Yes:** When creating a script that `az vm start` - and then proceeds with connecting to the vm with rdp and `Microsoft Remote Desktop` - **the vm is not ready**. This occurs even though the script is waiting for the `az vm start` process to finish first. The vm is not ready to be connected to and `Microsoft Remote Desktop` displays the following error message: 
> We couldn't connect to the remote PC. Make sure the PC is turned on and connected to the network, and that remote access is enabled.
Error code: 0x204

**Describe the solution you'd like**
Following the `az vm start` process - the vm should be connectable.

**Describe alternatives you've considered**
I have considered polling connection - or polling certain view information properties given from `az vm get-instance-view`. Or using a timeout. But these alternatives are not ideal.

**Additional context**
I have tried to `az vm wait --update` following the `az vm start` - but it won't wait longer, so the vm is still not ready to be connected to when the `az vm start` process is finished.

*My goal is to run a script that start vm and succesfully launches my vm with `Microsoft Remote Desktop`. Also, **note** that the connection problem is due to the vm not being ready - since it can be connected to 5 -20 secs after `az vm start`*.

**Comment by zikalino** [comment]
This is something that could be probably fixed in Azure CLI.
Probably also Python SDK could be a good place to look into.
However it needs some more investigation in Azure CLI side

**Comment by yonzhan** [comment]
add to S164.

**Comment by yungezz** [comment]
The behavior need to be improved at service side. `az vm start` call into service to get start status and return response to user.  When service return started state, the VM should be in ready state for user  connection. Suppose we do some thing in az cli to workaround the issue, what state should az cli poll/wait on? since service already returned started state.

**Comment by Lelelo1** [comment]
I don't know if something was changed recently - but it was **now possible**:
```
az vm start --ids $myid
```
and bash script is waiting - so that rest of the script successfully gives th tip adress of the machine, and machine being connectable:

**Comment by yungezz** [comment]
thanks @Lelelo1 for the update. We'll close the issue now then since your problem resolved. Pls feel free to create new issue if any other issue met.
